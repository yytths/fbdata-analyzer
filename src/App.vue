<template>
  <v-app>
    <v-app-bar
      app
      color="primary"
      dark
      dense
    >
      <div class="d-flex align-center">
        FBデータ解析ツール
      </div>
    </v-app-bar>

    <v-content>
      <v-container>
        <v-row>
          <v-textarea :rules="checkCounterPerRow()" v-model="inputFbData"/>
        </v-row>
        <v-row/>
        <v-row>
          <v-tabs grow>
            <v-tabs-slider></v-tabs-slider>
            <v-tab v-for="rt in Object.values(RECORD_TYPE)" :key="rt.label">
              {{ rt.label }}
            </v-tab>
            <v-tab-item v-for="rt in Object.values(RECORD_TYPE)" :key="rt.label">
              <app-header-record-data-table
                v-if="rt.value===RECORD_TYPE.HEADER.value" :items="headerRecords"
              />
              <app-header-record-data-table v-if="rt.value===RECORD_TYPE.DATA.value"/>
              <app-header-record-data-table v-if="rt.value===RECORD_TYPE.TRAILER.value"/>
              <app-header-record-data-table v-if="rt.value===RECORD_TYPE.END.value"/>
              <app-header-record-data-table v-if="rt.value===RECORD_TYPE.UNKNOWN.value"/>
            </v-tab-item>
          </v-tabs>
        </v-row>
      </v-container>
    </v-content>
  </v-app>
</template>

<script>
import appHeaderRecordDataTable from './components/AppHeaderRecordDataTable.vue';
import { RECORD_TYPE_OBJ } from './util/code';

export default {
  name: 'App',

  components: {
    appHeaderRecordDataTable,
  },

  data: () => ({
    myRecordType: '',
    inputFbData: '',
  }),

  computed: {
    RECORD_TYPE() {
      return RECORD_TYPE_OBJ;
    },

    headerRecords() {
      return this.inputFbData.split('\n').filter(((record) => record.startsWith('1')));
    },
  },

  methods: {
    checkCounterPerRow() {
      return [
        this.headerRecords.every((record) => record.length === 0 || record.length === 120)
          || 'ヘッダレコードが120文字ではありません。',
      ];
    },
  },
};
</script>
