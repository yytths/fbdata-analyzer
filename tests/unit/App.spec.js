import Vue from 'vue';
import Vuetify from 'vuetify';
import {
  mount,
  createLocalVue,
} from '@vue/test-utils';
import App from '@/App.vue';

Vue.use(Vuetify);
const localVue = createLocalVue(Vue);

describe('App.vue', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('normal test. layout', () => {
    const wrapper = mount(App, {
      localVue,
      vuetify,
    });
    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();

    const headerTitleElement = wrapper.find('header > div > div');
    expect(headerTitleElement.text()).toBe('FBデータ解析ツール');

    const textareaLabelElement = wrapper.find('#input-6')
    expect(textareaLabelElement.attributes().placeholder).toBe('FBデータを貼り付けてください');

    const tabsElements = wrapper.findAll('.v-slide-group__wrapper > div > div');
    expect(tabsElements.at(1).text()).toBe('ヘッダレコード');
    expect(tabsElements.at(1).classes()).toContain('v-tab--active');
    expect(tabsElements.at(2).text()).toBe('データレコード');
    expect(tabsElements.at(3).text()).toBe('トレーラレコード');
    expect(tabsElements.at(4).text()).toBe('エンドレコード');
    expect(tabsElements.at(5).text()).toBe('不明');
  });

});
