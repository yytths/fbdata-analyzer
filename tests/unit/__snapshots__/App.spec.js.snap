// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue normal test. layout 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-application--wrap">
    <header class="v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense v-app-bar v-app-bar--fixed primary" style="height: 48px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
      <div class="v-toolbar__content" style="height: 48px;">
        <div class="d-flex align-center">
          FBデータ解析ツール
        </div>
      </div>
    </header>
    <main class="v-content" style="padding: 48px 0px 0px 0px;">
      <div class="v-content__wrap">
        <div class="container">
          <div class="row">
            <div class="v-input v-textarea theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend><span>​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><textarea id="input-6" placeholder="FBデータを貼り付けてください" rows="5"></textarea></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="row"></div>
          <div class="row">
            <div class="v-tabs v-tabs--grow theme--light">
              <div role="tablist" class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text primary">
                <div class="v-slide-group__prev v-slide-group__prev--disabled">
                  <transition-stub name="fade-transition"></transition-stub>
                </div>
                <div class="v-slide-group__wrapper">
                  <div class="v-slide-group__content v-tabs-bar__content">
                    <div class="v-tabs-slider-wrapper" style="transition: none;">
                      <div class="v-tabs-slider"></div>
                    </div>
                    <div tabindex="0" aria-selected="true" role="tab" class="v-tab v-tab--active">
                      ヘッダレコード
                    </div>
                    <div tabindex="0" aria-selected="false" role="tab" class="v-tab">
                      データレコード
                    </div>
                    <div tabindex="0" aria-selected="false" role="tab" class="v-tab">
                      トレーラレコード
                    </div>
                    <div tabindex="0" aria-selected="false" role="tab" class="v-tab">
                      エンドレコード
                    </div>
                    <div tabindex="0" aria-selected="false" role="tab" class="v-tab">
                      不明
                    </div>
                  </div>
                </div>
                <div class="v-slide-group__next v-slide-group__next--disabled">
                  <transition-stub name="fade-transition"></transition-stub>
                </div>
              </div>
              <div class="v-window v-item-group theme--light v-tabs-items">
                <div class="v-window__container">
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
